### PR TITLE
Support weak form in elliptic DG

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

Almost gets the elliptic solver to work on large sphere domains! The attached `Lorentzian.yaml` configuration works, only BBH ID needs some more treatment (next PR).

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Elliptic/DiscontinuousGalerkin/Initialization.cpp [hunk 1]
```diff
@@ -74,9 +78,19 @@ void InitializeGeometry<Dim>::operator()(
   *inertial_coords =
       element_map->operator()(*logical_coords, 0., functions_of_time);
   // Jacobian
+  // Note: we can try to use `::dg::metric_identity_jacobian_quantities` here.
+  // When I tried (NV, Dec 2022) the DG residuals diverged on a sphere domain
```
Comment (nilsdeppe) [inline src/Elliptic/DiscontinuousGalerkin/Initialization.cpp]: Q: Dec. 2022 or Dec. 2023?

File: src/Elliptic/DiscontinuousGalerkin/DgOperator.hpp [hunk 2]
```diff
@@ -802,15 +825,17 @@ struct DgOperatorImpl<System, Linearized, tmpl::list<PrimalFields...>,
           get<Tags::NormalDotFluxForJump<PrimalMortarVars>>(
               remote_data.field_data)));
       primal_boundary_corrections_on_mortar *= penalty;
-      const auto add_remote_avg_contribution = [](auto& lhs, const auto& rhs) {
+      const auto add_avg_contribution = [](auto& lhs, const auto& rhs,
+                                           const double sign = 1.) {
         for (size_t i = 0; i < lhs.size(); ++i) {
-          lhs[i] += 0.5 * rhs[i];
+          lhs[i] += sign * 0.5 * rhs[i];
```
Comment (nilsdeppe) [inline src/Elliptic/DiscontinuousGalerkin/DgOperator.hpp]: [optional] consider precomputing `sign * 0.5` outside the loop. That multiplication is 1/3 the FLOPS in this expression. At least a few years ago, compilers weren't doing that automatically :(

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
LGTM! A couple suggestions. Feel free to squash & rebase right away!

File: src/Elliptic/DiscontinuousGalerkin/Initialization.cpp (hunk 1 above)
Comment (nilsvu) [reply src/Elliptic/DiscontinuousGalerkin/Initialization.cpp]: Oops 👍

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**